### DISCUSSION TITLE
What is best way to trigger something after the dialog is completely closed?

### DISCUSSION THREAD

**Discussion by hungdoansy** [radix-ui/primitives#2826] [Help]
Hi everyone,

Sorry if this question is too trivial. I've been working with dialogs for a while now. When the modal open state changes to `false`, there's a transition to make it disappear smoothly.

I'd like to update a state inside the dialog component after it's gone completely.

Is this a good way to do it? 
```js
useEffect(() => {
    if (!isOpen) {
        setTimeout(() => {
            setSomething(something)
        }, 200)
    }
}, [isOpen])
```

I assume 200ms is transition duration.

Thanks in advance.

**Comment by joaom00** [comment]
You can use `onCloseAutoFocus` will be called when the component is completely unmounted
